src/main/java/graphql/execution/incremental/IncrementalContext.java

```diff
@@ -0,0 +1,99 @@
+package graphql.execution.incremental;
+
+import graphql.Internal;
+import graphql.execution.reactive.SingleSubscriberPublisher;
+import graphql.incremental.DelayedIncrementalExecutionResult;
+import graphql.incremental.IncrementalPayload;
+import graphql.util.LockKit;
+import org.reactivestreams.Publisher;
+
+import java.util.Collection;
+import java.util.Collections;
+import java.util.Deque;
+import java.util.concurrent.ConcurrentLinkedDeque;
+import java.util.concurrent.atomic.AtomicBoolean;
+import java.util.concurrent.atomic.AtomicInteger;
+
+import static graphql.incremental.DelayedIncrementalExecutionResultImpl.newIncrementalExecutionResult;
+
+/**
+ * This provides support for @defer directives on fields that mean that results will be sent AFTER
+ * the main result is sent via a Publisher stream.
+ */
+@Internal
+public class IncrementalContext {
```

Comment:
The name doesnt quite right true to me - its not really a Context - but rather the state of calls - eg the tousranding ones etc...

Naming is hard

* IncrementalSupport
* IncrementalState
* IncrementalCallState

Reply:
I like `IncrementalCallState` - as it describes clearly what the class is responsible for.